graph/network/diffusion_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright ©2017 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package network
+
+import (
+	"math"
+	"sort"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/internal/ordered"
+	"gonum.org/v1/gonum/graph/simple"
+	"gonum.org/v1/gonum/mat"
+)
+
+var heatDiffusionTests = []struct {
```

Comment:
More test cases that came to my mind after some reading that you might want to add if it does not mean too much work:
* graph with isolated nodes
* graph with several components. In this case each component will have its own average.

Reply:
Yeah, those are the same class of test. I'll see what I can do to expand the test logic to handle those.

Reply:
I've gone the easy route and just added graphs of that form and tested the node values explicitly. The graphs are small enough that they are Obviously Correct™.